plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+                    strncasecmp(headers_extra->val.data, "gzip", 4) == 0) {
+                    flb_debug("[splunk_prot] body is gzipped");
+                    gzip_compressed = FLB_TRUE;
+                }
+            }
+        }
+    }
+
+    if (gzip_compressed == FLB_TRUE) {
+        ret = flb_gzip_uncompress((void *) request->data.data, request->data.len,
+                                  &gz_data, &gz_size);
+        if (ret == -1) {
+            flb_error("[elasticsearch_bulk_prot] gzip uncompress is failed");
+            return -1;
+        }
+
+        if (type == HTTP_CONTENT_JSON) {
+            parse_hec_payload_json(ctx, tag, gz_data, gz_size);
+        }
+
+        if (type == HTTP_CONTENT_TEXT) {
+            process_raw_payload_pack(ctx, tag, gz_data, gz_size);
+        }
+
+        if (type == HTTP_CONTENT_UNKNOWN) {
+            if (((char *)gz_data)[0] == '{') {
+                parse_hec_payload_json(ctx, tag, gz_data, gz_size);
+            }
+            else {
+                process_raw_payload_pack(ctx, tag, gz_data, gz_size);
+            }
+        }
+        flb_free(gz_data);
+    }
+    else {
+        if (type == HTTP_CONTENT_JSON) {
+            parse_hec_payload_json(ctx, tag, request->data.data, request->data.len);
+        }
+
+        if (type == HTTP_CONTENT_TEXT) {
+            process_raw_payload_pack(ctx, tag, request->data.data, request->data.len);
+        }
+
+        if (type == HTTP_CONTENT_UNKNOWN) {
+            if (request->data.data[0] == '{') {
+                parse_hec_payload_json(ctx, tag, request->data.data, request->data.len);
+            }
+            else {
+                process_raw_payload_pack(ctx, tag, request->data.data, request->data.len);
+            }
+        }
+    }
+
+    return 0;
+}
+
+static int process_hec_raw_payload(struct flb_splunk *ctx, struct splunk_conn *conn,
+                                   flb_sds_t tag,
+                                   struct mk_http_session *session,
+                                   struct mk_http_request *request)
+{
+    int type = -1;
+    struct mk_http_header *header;
+
+    header = &session->parser.headers[MK_HEADER_CONTENT_TYPE];
+    if (header->key.data == NULL) {
+        send_response(conn, 400, "error: header 'Content-Type' is not set\n");
+        return -1;
+    }
+
+    if (header->val.len == 10 &&
+        strncasecmp(header->val.data, "text/plain", 10) == 0) {
+        type = HTTP_CONTENT_TEXT;
+    }
+
+    if (type == -1) {
+        /* Not neccesary to specify content-type for Splunk HEC. */
+        flb_plg_debug(ctx->ins, "Mark as unknown type for ingested payloads");
+        type = HTTP_CONTENT_UNKNOWN;
+    }
+
+    if (request->data.len <= 0) {
+        send_response(conn, 400, "error: no payload found\n");
+        return -1;
+    }
+
+    /* Always handle as raw type of payloads here */
+    if (type == HTTP_CONTENT_TEXT || type == HTTP_CONTENT_UNKNOWN) {
```

Comment:
Isn't type either `HTTP_CONTENT_TEXT` or `HTTP_CONTENT_UNKNOWN` always if we get this far? if so then remove the conditional.